File: longest-consecutive-sequence-128/longest_consecutive_sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        l = len(nums)

        if l ==0:
            return 0
        

        num_set = set(nums)
       
        result = 1

        for num in num_set:
            
            if num-1 not in num_set:
                
                next_num = num+1
                while next_num in num_set:
                    next_num+=1
                    
                result = max(result, next_num-num)
        return result

        
        

File: longest-consecutive-sequence-128/longest_consecutive_sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    if (nums.length==0) {
        return 0;
    }

    const num_set = new Set(nums);
    let result = 1;

    for(num of num_set) {
        if(!num_set.has(num-1)){
            let next_num = num+1;
            while(num_set.has(next_num)){
                next_num++;

            }
            result = Math.max(result, next_num-num);
        }
    }

    return result;

};